### PR TITLE
New DataPoints filter for descriptor augmentation

### PR DESCRIPTION
**Pull request by boxanm**
This DataPointFilter adds a new descriptor with a given **name**, **dimension**, and **values** to the point cloud on which it is applied. It can be used, for example, when a map-building algorithm requires the scans to have the same descriptor as the existing map.

### CODE REVIEW

**Comment by boxanm** [comment]
I'm putting a **TODO** here to implement Python bindings for the filter ❌

File: pointmatcher/DataPointsFilters/AddDescriptor.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+// In-place filter
+template<typename T>
+void AddDescriptorDataPointsFilter<T>::inPlaceFilter(
+        DataPoints& cloud)
+{
+    Matrix matrix = PM::Matrix::Ones(descriptorDimension, cloud.getNbPoints());
+    for(std::size_t i = 0; i < descriptorDimension; ++i)
+    {
+        matrix.row(i) *= descriptorValues[i];
```
Comment (simonpierredeschenes) [inline pointmatcher/DataPointsFilters/AddDescriptor.cpp:35]: why is it a multiplicating instead of overwriting

**Review by simonpierredeschenes — CHANGES_REQUESTED** [review]
Good job, there is only one small comment to address

File: pointmatcher/DataPointsFilters/AddDescriptor.cpp (hunk 1 above)
Comment (boxanm) [reply pointmatcher/DataPointsFilters/AddDescriptor.cpp:35]: The Matrix `matrix` is initialized as Ones and each row is multiplicated by the corresponding descriptor value. Another option would be to assign `descriptorValues` to each column on `matrix`. Idk what's more optimal

**Pull request merged**